### PR TITLE
AST: GenericSignatures don't contain type variables

### PR DESCRIPTION
**Pull request by slavapestov**

### CODE REVIEW

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please test source compatibility

File: lib/AST/ASTContext.cpp [hunk 1]
```diff
@@ -1839,28 +1842,14 @@ void ASTContext::addLoadedModule(ModuleDecl *M) {
   getImpl().LoadedModules[M->getName()] = M;
 }
 
-static AllocationArena getArena(GenericSignature genericSig) {
-  if (!genericSig)
-    return AllocationArena::Permanent;
-
-  if (genericSig.hasTypeVariable()) {
-    assert(false && "Unsubstituted type variable leaked into generic signature");
```
Comment (CodaFi) [inline lib/AST/ASTContext.cpp:1847]: Do we preserve this assert anywhere else?

**Review by CodaFi — APPROVED** [review]

File: lib/AST/ASTContext.cpp (hunk 1 above)
Comment (slavapestov) [reply lib/AST/ASTContext.cpp:1847]: Yep, GenericSignature::get() on ASTContext.cpp line 4476. That seems to be the only entry point where we actually create new signatures, so I thought that was the best place to put it.

Comment (slavapestov) [reply lib/AST/ASTContext.cpp:1847]: I actually added that assert recently in 02db6324888bfb77a619b8f2433721e2f7ba0bf9, but kept the remaining logic around until I could do more tests. You re-worded it in 74b653518df3ac5304229b31b5ef0d2eb5f3db0f :-)

**Pull request merged**